update/update.go

```diff
@@ -74,6 +74,7 @@ func GetUpdateToolFromRepoCallback(toolName, version, repoName string) func() {
 		bin, err := gh.GetExecutableFromAsset()
 		if err != nil {
 			gologger.Fatal().Label("updater").Msgf("executable %v not found in release asset `%v` got: %v", toolName, gh.AssetID, err)
+			os.Exit(1)
```

Comment:
This code is unreachable since `gologger.Fatal()` already calls `os.Exit(..)` internally. I'm not sure if the wanted behavior wanted to be to return an error instead of aborting?

Speaking generally for future reference, let's reconsider carefully aborting execution from within a low-level library. Canonically the only allowed abortion in third-party libraries should be because of an unrecoverable internal panic.

Reply:
My bad; I probably didn't read it carefully. I will close the PR and rethink.